src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -136,4 +133,64 @@ void IDisposable.Dispose()
     {
         Navigation.LocationChanged -= HandleLocationChanged;
     }
+
+    bool ICascadingValueComponent.CanSupplyValue(Type valueType, string? valueName)
+    {
+        var formName = string.IsNullOrEmpty(valueName) ?
+            (_bindingContext?.Name) :
+            ModelBindingContext.Combine(_bindingContext, valueName);
+
+        if (_bindingInfo != null &&
+            string.Equals(_bindingInfo.FormName, formName, StringComparison.Ordinal) &&
+            _bindingInfo.ValueType.Equals(valueType))
+        {
+            // We already bound the value, but some component might have been destroyed and
+            // re-created. If the type and name of the value that we bound are the same,
+            // we can provide the value that we bound.
+            return true;
+        }
+
+        // Can't supply the value if this context is for a form with a different name.
+        if (FormValueSupplier.CanBind(formName!, valueType))
+        {
+            var bindingSucceeded = FormValueSupplier.TryBind(formName!, valueType, out var boundValue);
+            _bindingInfo = new BindingInfo(formName, valueType)
+            {
+                BindingResult = bindingSucceeded,
+                BoundValue = boundValue,
+            };
+
+            if (!bindingSucceeded)
+            {
+                // Report errors
+            }
+
+            return true;
+        }
+        
+        return false;
+    }
+
+    void ICascadingValueComponent.Subscribe(ComponentState subscriber)
+    {
+        throw new InvalidOperationException("Form values are always fixed.");
```

Comment:
I wonder if this would be a good use for the new `UnreachableException`?

Reply:
Hmm, I don't know the exact semantics of UnreachableException, it's not clear from looking at the doc comments, so I prefer we stick to InvalidOperationException as this is something we are doing elsewhere in the codebase.

Reply:
We can only do this if we can be completely sure that no other `[CascadingParameter]` is ever going to try to take a subscription on this (because the name or type happens to clash). Especially as the error message given here won't help people to debug the situation if that happens. We should be sure this is not possible at all for app developers except if they are doing extremely unusual things.

Reply:
We will filter this out within the `CanSupplyValue` call (or whatever is called). Mackinnon is working that part out.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/48451 to track.